shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXConnection.scala

```diff
@@ -146,10 +192,36 @@ class UCXConnection(peerExecutorId: Int, ucx: UCX) extends Connection with Loggi
     composeTag(composeUpperBits(peerClientId, bufferMsgType), bufferTag)
   }
 
-  private def composeTag(upperBits: Long, lowerBits: Long): Long = {
+  protected def composeRequestAmId(requestType: RequestType.Value): Int = {
+    val amId = requestType.id
+    if ((amId & 0x0000000F) != amId) {
+      throw new IllegalArgumentException(
+        s"Invalid request amId, it must be 4 bits: ${TransportUtils.toHex(amId)}")
+    }
+    amId
+  }
+
+  protected def composeRequestHeader(executorId: Long, txId: Long): Long = {
+    composeTag(executorId << 32, txId)
+  }
+
+  def extractExecutorId(header: Long): Long = {
+    (header & 0xFFFFFFFF00000000L) >> 32
+  }
+
+  def composeResponseAmId(requestType: RequestType.Value): Int = {
+    val amId = (0x00000010) | composeRequestAmId(requestType)
```

Comment:
can we get some named constants. Like "Executor_ID" mask for example.

Suggestion but not required.